src/asm/lexer.c

```diff
@@ -2049,20 +2138,16 @@ void lexer_CaptureMacroBody(char **capture, size_t *size)
 		do { /* Discard initial whitespace */
 			c = nextChar();
 		} while (isWhitespace(c));
-		/* Now, try to match either `REPT` or `ENDR` as a **whole** identifier */
-		if (startsIdentifier(c)) {
-			switch (readIdentifier(c)) {
+		/* Now, try to match either `label: MACRO` or `ENDM` as a **whole** identifier */
+		if (startsMacroWithinMacro(c)) {
```

Comment:
This looks bugged, since the function has some side-effects? This needs investigating.

Reply:
`startsMacroWithinMacro` has no side effects. `readMacroWithinMacro` does, but it's the same sort of side effect as the `nextChar()` and `shiftChars(1)` within the `lexer_CaptureRept` function: it skips ahead some more characters, but without the possibility of missing the corresponding `ENDR`/`ENDM`, which is all that matters since the body will be evaluated when it's actually used.

Reply:
Right, I see.